draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -768,24 +763,26 @@ guess. A new salt MUST be chosen for each claim.
 
 The RECOMMENDED minimum length of the randomly-generated portion of the salt is 128 bits.
 
-Note that minimum 128 bits would be necessary when SHA-256, HMAC-SHA256, or a function of similar strength is used, but a smaller salt size might achieve similar level of security if a stronger iterative derivation function is used.
+Note that minimum 128 bits would be necessary when SHA-256, or a function of similar strength is used.
 
 The Issuer MUST ensure that a new salt value is chosen for each claim,
 including when the same claim name occurs at different places in the
 structure of the SD-JWT. This can be seen in Example 3 in the Appendix,
 where multiple claims with the name `type` appear, but each of them has
 a different salt.
 
-## Choice of a digest derivation algorithm
+## Choice of a Hash Algorithm
 
-For the security of this scheme, the digest derivation algorithm is required to be preimage and collision
+For the security of this scheme, the hash algorithm is required to be preimage and collision
 resistant, i.e., it is infeasible to calculate the salt and claim value that result in
 a particular digest, and it is infeasible to find a different salt and claim value pair that
 result in a matching digest, respectively.
 
 Furthermore the hash algorithms MD2, MD4, MD5, RIPEMD-160, and SHA-1
 revealed fundamental weaknesses and they MUST NOT be used.
 
+Note that implementations willing to use functions other than hash functions, for example to achieve a smaller salt size with similar level of security by using a stronger iterative derivation function, are RECOMMENDED to define a profile of this specification, i.e., when an HMAC or another type of hash function is used for digest calculation, it needs to be defiend how a secret cryptographic key or other cryptographic secret is can be used instead of a salt value.
```

Comment:
Technically speaking, that would not be a profile, as it would need to change normative things within this spec. We cannot really give this recommendation here. All we can say is: HMAC or similar is at your own risk, non-standard, etc.; I'm not sure if we want to say that here.

Reply:
I think what we are trying to say here is that a smaller salt could potentially be used in conjunction with a more computationally expensive hash function. Where a hash function is a single input -> single digest output. And the  Hash Function Claim section says sd_hash_alg can be "a value defined in another specification and/or profile of this specification". Such a spec/profile would need to define that hash function for use at the SD-JWT layer as a single input to single digest output thing. But it could map/profile something like PBKDF2 to do that or define a 30 billion iteration sha2 thing. 

I'm not sure how much of that we want to say here. But it would be an "at your own risk" kind of thing. 

I think that the specific talk of HMAC and cryptographic secrets here is confusing things though. It certainly confused me :)

Reply:
> the Hash Function Claim section says sd_hash_alg can be "a value defined in another specification and/or profile of this specification"
personally, I am ok removing this section, with this PR, don't think it's relevant anymore.

Reply:
also removed this clause to prevent confusion.

Reply:
cc @christianpaquin for transparency that during IETF115 OAuth WG, there was a decision to revert to hashing alg because including HMAC includes a lot of additional definitions to be made, not just saying _sd_hash_alg can be HMAC, and we should keep SD-JWT spec simple.